lib/wibox/widget/imagebox.lua

```diff
@@ -79,36 +75,38 @@ end
 --- Set an imagebox' image
 -- @property image
 -- @param image Either a string or a cairo image surface. A string is
---   interpreted as the path to a png image file.
+--   interpreted as the path to a image file.
 -- @return true on success, false if the image cannot be used
-
 function imagebox:set_image(image)
     if type(image) == "string" then
-        image = surface.load(image)
-        if not image then
-            print(debug.traceback())
-            return false
+        -- try to load rsvg handle from file if it's vector
+        local is_vector = string.match(image, "%.svg$")
+        self._private.handle = is_vector and load_svg_and_handle_errors(image) or nil
+
+        if not self._private.handle then
+            -- try to load cairo surface from raster image file
+            local _surface = surface.load(image)
+            if not _surface or _surface.width <= 0 or _surface.height <= 0 then
+                return false
+            end
+            self._private.image = _surface
+        else
+            -- set dummy surface object for default sizes reference
+            local dim = self._private.handle:get_dimensions()
+            self._private.image = { width = dim.width, height = dim.height }
```

Comment:
Given they are different types, what about using 2 different attribute variables? This would avoid some type confusion issues down the line? This will require to cleanup both, but that's less effort than adding `if` elsewhere.